chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -1503,38 +1482,32 @@ impl Actor for PeerManagerActor {
         if let Some(server_addr) = self.config.addr {
             // TODO: for now crashes if server didn't start.
 
-            ctx.spawn(TcpListener::bind(server_addr).into_actor(self).then(
-                move |listener, act, ctx| {
-                    let listener = listener.unwrap();
-                    let incoming = IncomingCrutch {
-                        listener: tokio_stream::wrappers::TcpListenerStream::new(listener),
-                    };
-                    info!(target: "stats", "Server listening at {}@{}", act.my_peer_id, server_addr);
-                    let pending_incoming_connections_counter =
-                        act.pending_incoming_connections_counter.clone();
-                    let peer_counter = act.peer_counter.clone();
-                    let max_num_peers: usize = act.config.max_num_peers as usize;
-
-                    ctx.add_message_stream(incoming.filter_map(move |conn| {
-                        if let Ok(conn) = conn {
-                            if pending_incoming_connections_counter.load(Ordering::SeqCst)
-                                + peer_counter.load(Ordering::SeqCst)
-                                < max_num_peers + LIMIT_PENDING_PEERS
-                            {
-                                pending_incoming_connections_counter.fetch_add(1, Ordering::SeqCst);
-                                return future::ready(Some(
-                                    PeerManagerMessageRequest::InboundTcpConnect(
-                                        InboundTcpConnect::new(conn),
-                                    ),
-                                ));
-                            }
-                        }
+            debug!(target: "network", message = "starting server", at = ?server_addr);
+            let address = ctx.address();
 
-                        future::ready(None)
-                    }));
-                    actix::fut::ready(())
-                },
-            ));
+            async move {
+                let listener = TcpListener::bind(server_addr).await.ok();
+                let listener = if let Some(listener) = listener {
+                    listener
+                } else {
+                    return;
+                };
+
+                loop {
+                    let (conn, client_addr) =
+                        if let Ok((conn, client_addr)) = listener.accept().await {
+                            (conn, client_addr)
+                        } else {
+                            return;
```

Comment:
so any error on the listener part will cause us to stop listening at all?

or is there some mechanism that calls 'started' again?

Reply:
Currently, we continue on error. I changed the logic to be the same as before.

We will continue listening, and ignore all failures of `listener.accept()`. Same as before.